src/connection.rs

```diff
@@ -354,19 +349,15 @@ impl<T: AsyncRead + AsyncWrite + Unpin> Connection<T> {
             }
         }
 
-        if let Err(ConnectionError::Closed) = result {
-            return Ok(None)
-        }
-
-        result
+        result.map(move |stream| (stream, self))
```

Comment:
`result` is always `Err` here right, so the `map` just aligns the types? Presumable you want to avoid increasing the indentation of all the above code up to `self.is_closed = true` if it were put in an `Err(e) => ` branch of a `match` on the `result`?

Reply:
I am fine with using a `match` instead. 99ec2b0 reformats the code accordingly.